File: interpreter/src/pregen/scala/org/enso/interpreter/nodes/expression/operator/MultiplyOperatorNode.scala
package org.enso.interpreter.nodes.expression.operator

import com.oracle.truffle.api.dsl.Specialization
import com.oracle.truffle.api.nodes.NodeInfo
import org.enso.interpreter.nodes.expression.BinaryOperatorNode

@NodeInfo(shortName = "*")
abstract class MultiplyOperatorNode extends BinaryOperatorNode {

  @Specialization
  protected def multiply(left: Long, right: Long): Long =
    left * right
}

File: interpreter/src/main/java/org/enso/interpreter/nodes/expression/operator/MultiplyOperatorNode.java
package org.enso.interpreter.nodes.expression.operator;

import com.oracle.truffle.api.dsl.Specialization;
import com.oracle.truffle.api.nodes.NodeInfo;
import org.enso.interpreter.nodes.expression.BinaryOperatorNode;

@NodeInfo(shortName = "*")
public abstract class MultiplyOperatorNode extends BinaryOperatorNode {

    @Specialization
    protected long multiply(long left, long right) {
        return left * right;
    }
}
